src/Entity/OgRole.php

```diff
@@ -177,8 +177,39 @@ public function setRoleType($role_type) {
   /**
    * {@inheritdoc}
    */
+  public function getName() {
+    // If the name is not set yet, try to derive it from the ID.
+    if (empty($this->name) && !empty($this->id())) {
+      list(, , $name) = explode('-', $this->id());
```

Comment:
There should always be three elements in the current implementation. But I think you're on to something, this is indeed a bug. I don't think there is anything stopping people to use dashes in their entity type IDs and bundle IDs, which means that counting the dashes is not going to cut it.

I'll change it to match exactly `"{$this->getGroupEntityTypeId()}-{$this->getGroupBundleId()}-$role_name"`, then we'll eliminate any dash-related problems.